feature/preview/src/main/java/com/google/jetpackcamera/feature/preview/PreviewViewModel.kt

```diff
@@ -259,6 +326,21 @@ class PreviewViewModel @AssistedInject constructor(
         }
     }
 
+    fun showSnackBarForDisabledHdrToggle(disabledReason: CaptureModeToggleUiState.DisabledReason) {
+        viewModelScope.launch {
+            _previewUiState.update { old ->
+                (old as? PreviewUiState.Ready)?.copy(
+                    snackBarToShow = SnackbarData(
+                        cookie = "DisabledHdrToggle",
```

Comment:
nit: this cookie should be unique (see how it is done for `captureImageInternal`). You could rename `imageCaptureCalledCount` to `snackbarCount` and reuse that if you want.